\documentclass{beamer}
\usepackage{amsmath}
\usepackage{booktabs}
\usepackage{units}
\usepackage[latin1]{inputenc}
\usefonttheme[onlylarge]{structurebold}
\setbeamerfont*{frametitle}{size=\normalsize,series=\bfseries}
\setbeamertemplate{navigation symbols}{}
\usetheme{Goettingen}
\defbeamertemplate{itemize item}{checkbox}{\Square}
\defbeamertemplate{itemize item}{checked}{\Square\llap\CheckmarkBold}

\title[Lab Meeting IV]{Lab Meeting IV}
\author{Carles Boix}
\begin{document}

% DO some background on RAP1
\begin{frame}
 \begin{center}
        {\large \textsc{Analysis of RNA-seq data for RAP1:}}
    \end{center}

    \textbf{RAP1:}
    \begin{itemize}
        \item General Regulatory Factor\footnote{Fourel et al.,2002}
            \begin{itemize}
                \item Abundant + essential.
                \item Binding sites found in many promoters.
                \item Enhancers of regulation.
            \end{itemize}
        \item According to SGD, targets 1655 genes
        \item Chromatin silencing\footnote{Konig et al., 1996}
        \item Regulation of ribosome components\footnote{Lieb et al., 2001}
    \end{itemize}
\end{frame}
\begin{frame}
    \begin{center}
        {\Large \textsc{First Disclaimer:}}
    \end{center}
    RNA-seq data for only the ATFs for RAP1 was generated during the first 3 weeks of classes.
\end{frame}

\begin{frame}
    \begin{center}
        {\Large \textsc{Second Disclaimer:}}
    \end{center}
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.95\textwidth]{../FACShist.png}
        \label{fig:facs}
    \end{figure}
    \begin{center}
        Number of GFP positive cells by FACS
    \end{center}

\end{frame}

\begin{frame}
    CuffDiff run to compare:
    \begin{itemize}
        \item 0'' to 15''
        \item 0'' to 90''
        \item 0''   min to 90'' uninduced
        \item 15'' to 90''
        \item 15'' to 90'' uninduced
        \item 90'' to 90'' uninduced
    \end{itemize}
    \bigskip
    \pause
    The only significant differences found were between \textbf{0 and 90'' induced}.
\end{frame}

\begin{frame}
    \begin{table}[ht]
        \centering
        \begin{tabular}{rlrr}
            \toprule
            & Gene & Log2 Fold Change & p-value \\ 
            \midrule
            1 & TIS11 & 2.69 & $5 \times 10^{-5}$ \\ 
            2 & INO1 & 2.50 & $5 \times 10^{-5}$\\ 
            3 & SSA2 & -1.90 & $5 \times 10^{-5}$ \\ 
            4 & YEL073C & 1.74 & $5 \times 10^{-5}$ \\ 
            5 & GPG1 & 1.52 & $5 \times 10^{-5}$ \\ 
            6 & ICY1 & 1.39 & $5 \times 10^{-5}$ \\ 
            7 & MRS3 & 1.35 & $5 \times 10^{-5}$ \\ 
            8 & YOR387C & 1.27 & $5 \times 10^{-5}$ \\ 
            9 & PDR12 & 1.26 & $5 \times 10^{-5}$ \\ 
            10 & GSC2 & 1.25 & $5 \times 10^{-5}$ \\ 
            11 & YMR090W & 1.24 & $5 \times 10^{-5}$ \\ 
            \bottomrule
        \end{tabular}
    \end{table}
    \pause
    Rap1 has 1655 targets, according to the SGD. 
\end{frame}


\begin{frame}
    \begin{table}[ht]
        \centering
        \begin{tabular}{rlrr}
            \toprule
            & Gene & Log2 Fold Change & p-value \\ 
            \midrule
            1 & \textbf{\color{red}{TIS11}} & 2.69 & $5 \times 10^{-5}$ \\ 
            2 & INO1 & 2.50 & $5 \times 10^{-5}$\\ 
            3 & \textbf{\color{red}{SSA2}} & -1.90 & $5 \times 10^{-5}$ \\ 
            4 & \textbf{\color{red}{YEL073C}} & 1.74 & $5 \times 10^{-5}$ \\ 
            5 & GPG1 & 1.52 & $5 \times 10^{-5}$ \\ 
            6 & ICY1 & 1.39 & $5 \times 10^{-5}$ \\ 
            7 & \textbf{\color{red}{MRS3}} & 1.35 & $5 \times 10^{-5}$ \\ 
            8 & \textbf{\color{red}{YOR387C}} & 1.27 & $5 \times 10^{-5}$ \\ 
            9 & \textbf{\color{red}{PDR12}} & 1.26 & $5 \times 10^{-5}$ \\ 
            10 & \textbf{\color{red}{GSC2}} & 1.25 & $5 \times 10^{-5}$ \\ 
            11 & YMR090W & 1.24 & $5 \times 10^{-5}$ \\ 
            \bottomrule
        \end{tabular}
    \end{table}
    Rap1 has 1655 targets, according to the SGD. ($p = 0.14$)
\end{frame}

\begin{frame}
    \textbf{TIS11:}
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../TIS11.png}
        \label{fig:tis11}
    \end{figure}
    \textbf{GSC2}
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../GSC2_0-90i.png}
        \label{fig:gsc2}
    \end{figure}
\end{frame}

\begin{frame}
    \begin{figure}[ht!]
            \centering
            \includegraphics[width=.8\textwidth]{../FoldChangeATAR.png}
            \caption{0 - 90 minutes induced. Distribution of fold change for {\color{gray} all genes}, {\color{blue} SGD targets}.}
            \label{fig:all.tar}
        \end{figure}
    Wilcoxon significance:
    \begin{itemize}
        \item \textbf{All vs. SGD targets:} p-value $= 0.2101$.
    \end{itemize}
    \emph{There is no difference in fold change between Rap1 targets and the full set of genes.}
\end{frame}

\begin{frame}
    \textbf{Yarragudi et al, 2007} - provides a list of probable and putative Rap1 targets by experimentation with Rap1 ts mutants.
    \begin{figure}[ht!]
            \centering
            \includegraphics[width=.8\textwidth]{../FoldChangeAPrPu.png}
            \caption{0 - 90 minutes induced. Distribution of fold change for {\color{gray} all genes}, {\color{blue} putative direct Rap1 targets}, and {\color{red} probable direct Rap1 targets} from Yarragudi et al, 2007.}
            \label{fig:targ}
        \end{figure}
    Wilcoxon significance:
    \begin{itemize}
        \item \textbf{All vs. Putative:} p-value $= 0.5316$.
        \item \textbf{All vs. Probable:} p-value $= 0.08193$.
    \end{itemize}
\end{frame}

\begin{frame}
    \emph{Where do the top hits come from?}
    \pause
    \begin{table}[ht]
        \centering
        \begin{tabular}{rlr}
            \toprule
            & Gene & Function \\ 
            \midrule
            1 & \textbf{\color{red}{TIS11}} & Increases on DNA replication stress \\ 
            2 & INO1 & Inositol phosphate synthesis \\ 
            3 & \textbf{\color{red}{SSA2}} & Stress response / Protein folding \\ 
            4 & \textbf{\color{red}{YEL073C}} & Unknown function, regulated by Inositol  \\ 
            5 & GPG1 & Signal Transduction \\ 
            6 & ICY1 & Unknown function, paralog to \textbf{\color{red}{ICY2}} \\ 
            7 & \textbf{\color{red}{MRS3}} & Iron transporter (mitochondrion) \\ 
            8 & \textbf{\color{red}{YOR387C}} &  Unknown function, has paralog VEL1\\ 
            9 & \textbf{\color{red}{PDR12}} &  Membrane transporter\\ 
            10 & \textbf{\color{red}{GSC2}} &  Spore wall formation\\ 
            11 & YMR090W & Unknown function \\ 
            \bottomrule
        \end{tabular}
    \end{table}
\end{frame}


% SHOW GO TERMs for 0-90 
\begin{frame}

    \textsc{Enriched GO terms, 0 to 90'' induced:}
    \begin{table}[ht]
        \centering
        \scriptsize
        \begin{tabular}{lr}
        \toprule
        \textbf{GO term} &\textbf{p-value} \\ 
        \midrule
        RNA modification guide activity & $8.194 \times 10^{-09}$ \\
        rRNA modification guide activity & $8.194 \times 10^{-09}$ \\
        base pairing with rRNA & $1.228 \times 10^{-08}$ \\
        rRNA modification & $1.719 \times 10^{-07}$ \\
        small nucleolar ribonucleoprotein complex & $1.609 \times 10^{-06}$ \\
        \bottomrule
    \end{tabular}

    \end{table}

    \vfill
    \relax
    \vskip 100pt
    \scriptsize
    \rule{\linewidth}{0.5pt}
    {FuncAssociate \hfill Berriz, 2009}
\end{frame}
% SHOW GO TERMs for 0-15

\begin{frame}
    \textsc{Enriched GO terms, 0 to 15'' induced:}

    \begin{table}[ht]
    \centering
    \scriptsize
        \begin{tabular}{lr}
        \toprule
        \textbf{GO term} & \textbf{p-value}\\ 
        \midrule
        cytosolic large ribosomal subunit & $3.1129 \times 10^{-37}$ \\
        cytosolic small ribosomal subunit & $8.9788 \times 10^{-24}$ \\
        cytosolic ribosome & $9.6384 \times 10^{-57}$ \\
        rRNA export from nucleus & $3.4673 \times 10^{-07}$ \\
        rRNA transport & $3.4673 \times 10^{-07}$ \\
        glycolysis & $2.1990 \times 10^{-09}$ \\
        cytoplasmic translation & $5.8050 \times 10^{-44}$ \\
        cytosolic part & $1.3412 \times 10^{-43}$ \\
        glucose catabolic process & $1.6974 \times 10^{-10}$ \\
        structural constituent of ribosome & $1.5817 \times 10^{-31}$ \\
        large ribosomal subunit & $1.8137 \times 10^{-19}$ \\
        hexose catabolic process & $5.4872 \times 10^{-09}$ \\
        ribosomal subunit & $6.0681 \times 10^{-31}$ \\
        monosaccharide catabolic process & $4.4285 \times 10^{-09}$ \\
        small ribosomal subunit & $1.9764 \times 10^{-12}$ \\
        ribosome & $1.9820 \times 10^{-27}$ \\
        pyridine nucleotide metabolic process & $4.3267 \times 10^{-05}$ \\
        glucose metabolic process & $2.7083 \times 10^{-06}$ \\
        structural molecule activity & $3.7227 \times 10^{-14}$ \\
        cytosol & $2.2175 \times 10^{-12}$ \\
        translation & $2.8191 \times 10^{-09}$ \\
        \bottomrule
        \end{tabular}
    \end{table}
    
    \scriptsize
    \rule{\linewidth}{0.5pt}
    {FuncAssociate \hfill Berriz, 2009}
\end{frame}

\begin{frame}
    \textsc{Enriched GO terms, 0 to 15'' induced:}

    \begin{table}[ht]
    \centering
    \scriptsize
        \begin{tabular}{lr}
        \toprule
        \textbf{GO term} & \textbf{p-value}\\ 
        \midrule
        \textbf{\color{blue}{cytosolic large ribosomal subunit}} & $3.1129 \times 10^{-37}$ \\
        \textbf{\color{blue}{cytosolic small ribosomal subunit}} & $8.9788 \times 10^{-24}$ \\
        \textbf{\color{blue}{cytosolic ribosome}} & $9.6384 \times 10^{-57}$ \\
        \textbf{\color{cyan}{rRNA export from nucleus}} & $3.4673 \times 10^{-07}$ \\
        \textbf{\color{cyan}{rRNA transport}} & $3.4673 \times 10^{-07}$ \\
        glycolysis & $2.1990 \times 10^{-09}$ \\
        \textbf{\color{cyan}{cytoplasmic translation}} & $5.8050 \times 10^{-44}$ \\
        cytosolic part & $1.3412 \times 10^{-43}$ \\
        glucose catabolic process & $1.6974 \times 10^{-10}$ \\
        \textbf{\color{blue}{structural constituent of ribosome}} & $1.5817 \times 10^{-31}$ \\
        \textbf{\color{blue}{large ribosomal subunit}} & $1.8137 \times 10^{-19}$ \\
        hexose catabolic process & $5.4872 \times 10^{-09}$ \\
        \textbf{\color{blue}{ribosomal subunit}} & $6.0681 \times 10^{-31}$ \\
        monosaccharide catabolic process & $4.4285 \times 10^{-09}$ \\
        \textbf{\color{blue}{small ribosomal subunit}} & $1.9764 \times 10^{-12}$ \\
        \textbf{\color{blue}{ribosome}} & $1.9820 \times 10^{-27}$ \\
        pyridine nucleotide metabolic process & $4.3267 \times 10^{-05}$ \\
        glucose metabolic process & $2.7083 \times 10^{-06}$ \\
        structural molecule activity & $3.7227 \times 10^{-14}$ \\
        cytosol & $2.2175 \times 10^{-12}$ \\
        \textbf{\color{cyan}{translation}} & $2.8191 \times 10^{-09}$ \\
        \bottomrule
        \end{tabular}
    \end{table}

    \scriptsize
    \rule{\linewidth}{0.5pt}
    {FuncAssociate \hfill Berriz, 2009}
    % Results come from FuncAssociate.
\end{frame}

% \begin{frame}
%     Some of the top targets were $SNR$ (small nucleolar RNAs), which aid in methylation, etc. These do not show a particular pattern, but have a large variance:
%         \begin{figure}[ht!]
%             \centering
%             \includegraphics[width=.8\textwidth]{../FoldChangeASNR.png}
%             \caption{0 - 90 minutes induced. Distribution of fold change for {\color{gray} all genes}, {\color{red} all SNR}}
%             \label{fig:all.snr}
%         \end{figure}

%     Wilcoxon significance:
%     \begin{itemize}
%         \item \textbf{All vs. SNR:} p-value $= 0.2257$.
%     \end{itemize}
% \end{frame}

\begin{frame}
    \textbf{Lieb et al., 2001} - almost all ($\nicefrac{124}{137}$) ribosomal protein genes (RPG) have binding sites for Rap1.\\

{ \scriptsize Using $\sim 300$ genes classified under the ribsome cell component in gene ontology:}
    \begin{figure}[ht!]
            \centering
            \includegraphics[width=.8\textwidth]{../FoldChangeARIB2.png}
            \caption{0 - 15 minutes induced. Distribution of fold change for {\color{gray} all genes}, {\color{blue} all ribosome genes}}
            \label{fig:all.rib}
        \end{figure}

        RPGs seem to be downregulated by overexpression of Rap1.

 Wilcoxon significance:
    \begin{itemize}
        \item \textbf{All vs. Ribosomal} p-value $= 2.2 \times 10^{-16}$
    \end{itemize}
    % Means: 0.02466845 vs. -0.1021341
\end{frame}

\begin{frame}
    \textbf{Lieb et al., 2001} - almost all ($\nicefrac{124}{137}$) ribosomal protein genes (RPG) have binding sites for Rap1.\\

{ \scriptsize Using $\sim 300$ genes classified under the ribsome cell component in gene ontology:}
    \begin{figure}[ht!]
            \centering
            \includegraphics[width=.8\textwidth]{../FoldChangeARIB.png}
            \caption{0 - 90 minutes induced. Distribution of fold change for {\color{gray} all genes}, {\color{blue} all ribosome genes}}
            \label{fig:all.rib}
        \end{figure}

        RPGs seem to be downregulated by overexpression of Rap1.

 Wilcoxon significance:
    \begin{itemize}
        \item \textbf{All vs. Ribosomal} p-value $=7.899 \times 10^{-16}$
    \end{itemize}
    % Means: 0.02466845 vs. -0.1021341
\end{frame}

\begin{frame}

    \bigskip
    { \scriptsize Using a different transcription factor (MIG1):}
    \begin{figure}[ht!]
            \centering
            \includegraphics[width=.8\textwidth]{../FoldChangeAMIGRIB2.png}
            \caption{0 - 15 minutes induced for MIG1. Distribution of fold change for {\color{gray} all genes}, {\color{blue} all ribosome genes}}
            \label{fig:all.rib}
        \end{figure}

 Wilcoxon significance:
    \begin{itemize}
        \item \textbf{MIG1 all vs. MIG1 ribosomal} p-value $= 2.2 \times 10^{-16}$
    \end{itemize}
    % Means: 0.02466845 vs. -0.1021341
\end{frame}

\begin{frame}
    
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../MIGvRAPRIB2.png}
        \label{fig:mig.rap.rib}
    \end{figure}\footnotesize

    {\footnotesize \emph{MIG1 v. RAP1 0-15 Ribosome} \hfill Adjusted $R^2 = 0.7816$.}
\end{frame}



% A case that I showed wrong was Tomar, 2008. (RNR + SNF / SWI)

\begin{frame}
    Genes w/ low p-values from 0 to 15 minutes (\emph{immediate response}) are highly enriched for stress response: \emph{DDR2}, \emph{SSA2}, \emph{PHM7}, \emph{TMA10}, \emph{HSP26}\dots\\
    \bigskip
    \pause
    \emph{Why might the yeast stress response be activated?}
    \begin{itemize}
        \item[\emph{(1)}] \textbf{Freeman et al., 1995} - overexpression of Rap1 in yeast is toxic. 
        \item[\emph{(2)}] The ATF-EV construct may be toxic. 
        \item[\emph{(3)}] \textbf{Banerjee et al., 2004} show that progesterone activates the stress response (however, $\beta$-estradiol is native, progesterone is not).
    \end{itemize}

    $\nicefrac{68}{152}$ genes under ``response to stress'' are also Rap1 Targets. (p-val = $0.00136$ by $\chi^2$)
    % OR MAYBE of our ATF is toxic!?
\end{frame}

\begin{frame}
    %A comparison of the 90 induced vs. 90 uninduced shows no strong changes in patterns in expression.\\
    \emph{Is the stress response activated?}\\

    {\scriptsize Look at 0 to 15 minutes induction. The $ \sim 150$ genes annotated under GO term \emph{response to stress} are, on average, upregulated between 0 and 15 min.}\\
    
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../FoldChangeASTR.png}
        \caption{0 - 15 minutes induced. Distribution of fold change for {\color{gray} all genes}, {\color{blue} stress response genes}.}
        \label{fig:targ2}
    \end{figure}

Wilcoxon significance:
    \begin{itemize}
        \item \textbf{All vs. Stress Response:} p-value $= 3.031 \times 10^{-08}$
    \end{itemize}
\end{frame}

\begin{frame}
    %A comparison of the 90 induced vs. 90 uninduced shows no strong changes in patterns in expression.\\
    \emph{Is the stress response a $\beta$-estradiol or otherwise technical response?}\\

    {\scriptsize 0 to 15 minutes induction for another transcription factor (MIG1). }\\
    
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../FoldChangeAMIGSTR.png}
        \caption{0 - 15 minutes induced for MIG1. Distribution of fold change for {\color{gray} all genes}, {\color{blue} stress response genes}.}
        \label{fig:targ4}
    \end{figure}

Wilcoxon significance:
    \begin{itemize}
        \item \textbf{MIG1 All vs. MIG1 Stress Response:} p-value $= 9.934 \times 10^{-08}$
    \end{itemize}
\end{frame}

\begin{frame}
    
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../MIGvRAPSTR.png}
        \label{fig:targ5}
    \end{figure}


    {\footnotesize \emph{MIG1 v. RAP1 0-15 Stress Response} \hfill Adjusted $R^2 = 0.8914$.}
\end{frame}


\begin{frame}
    
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../MIGvRAPALL.png}
        \label{fig:mig.rap.rib}
    \end{figure}

    {\footnotesize \emph{MIG1 v. RAP1 0-15 All} \hfill Adjusted $R^2 = 0.6428$.}
\end{frame}

\begin{frame}
    \begin{figure}[ht!]
        \centering
        \includegraphics[width=.8\textwidth]{../015090RAPALL.png}
        \label{fig:mig.rap.rib}
    \end{figure}

    {\footnotesize \emph{RAP1 0 to 15 v. 0 to 90 All} \hfill Adjusted $R^2 = 0.09552$.}
\end{frame}


\begin{frame}
    \begin{center}
        {\large \textsc{Conclusions}}
    \end{center}

    \begin{itemize}
        \item Clear fast stress response after adding $\beta$-estradiol.
        \item Ribosomal down-regulation in contradicts what we know about Rap1 targeting.
        \item Stress may also lead to decreased ribosome expression.
        \item Coordinated response after 15 minutes in both MIG1 and RAP1.
            \bigskip
        \item ATF may not have worked.
        \item Because Rap1 is a GRF, it may need TF partners to activate/repress.
    \end{itemize}
    % unlikely given 1655 targets!!!!!
    % FIND examples.
    % ALSO we may not have enough time points to do anything about it (save this for later??)
\end{frame}


\end{document}
